experimental/components/VideoPlayer/library/src/desktopMain/kotlin/org/jetbrains/compose/videoplayer/DesktopVideoPlayer.kt

```diff
@@ -1,63 +1,137 @@
 package org.jetbrains.compose.videoplayer
 
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.DisposableEffect
-import androidx.compose.runtime.SideEffect
-import androidx.compose.runtime.remember
+import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.awt.SwingPanel
 import androidx.compose.ui.graphics.Color
+import kotlinx.coroutines.delay
 import uk.co.caprica.vlcj.factory.discovery.NativeDiscovery
 import uk.co.caprica.vlcj.player.base.MediaPlayer
+import uk.co.caprica.vlcj.player.base.MediaPlayerEventAdapter
 import uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent
 import uk.co.caprica.vlcj.player.component.EmbeddedMediaPlayerComponent
+import uk.co.caprica.vlcj.player.embedded.EmbeddedMediaPlayer
+import java.awt.Component
 import java.util.*
+import kotlin.math.roundToInt
 
 @Composable
-internal actual fun VideoPlayerImpl(url: String, width: Int, height: Int) {
-    NativeDiscovery().discover()
-    val mediaPlayerComponent = remember {
-        // see https://github.com/caprica/vlcj/issues/887#issuecomment-503288294 for why we're using CallbackMediaPlayerComponent for macOS.
-        if (isMacOS()) {
-            CallbackMediaPlayerComponent()
-        } else {
-            EmbeddedMediaPlayerComponent()
+internal actual fun VideoPlayerImpl(
+    url: String,
+    isResumed: Boolean,
+    volume: Float,
+    speed: Float,
+    seek: Float,
+    isFullscreen: Boolean,
+    modifier: Modifier,
+    onFinish: (() -> Unit)?
+): State<Progress> {
+    val mediaPlayerComponent = initializeMediaPlayerComponent()
```

Comment:
It used to be "remembered"
`val mediaPlayerComponent = remember {`

I think that "remember" is still needed here. Was it removed accidentaly?

Reply:
I had moved it inside `initializeMediaPlayerComponent()` but now changed it.